runtime/jslib.js

```diff
@@ -394,76 +388,55 @@ function caml_js_wrap_callback(f) {
 //Provides: caml_js_wrap_callback_arguments
 //Requires: caml_callback
 function caml_js_wrap_callback_arguments(f) {
-  return function () {
-    var len = arguments.length;
-    var args = new Array(len);
-    for (var i = 0; i < len; i++) args[i] = arguments[i];
-    return caml_callback(f, [args]);
-  };
+  return (...args) => caml_callback(f, [args]);
 }
 //Provides: caml_js_wrap_callback_strict const
 //Requires: caml_callback
 function caml_js_wrap_callback_strict(arity, f) {
-  return function () {
-    var n = arguments.length;
-    var args = new Array(arity);
-    var len = Math.min(arguments.length, arity);
-    for (var i = 0; i < len; i++) args[i] = arguments[i];
+  return (...args) => {
+    args.length = arity;
```

Comment:
You can use this trick to change the length of an array. I will comment the benchmark later.

Reply:
Oops, `splice` is better than `length`: https://jsben.ch/nRrGR

Reply:
https://jsben.ch/dsvFX